lib/transmuxer.js

```diff
@@ -993,11 +1036,61 @@ VideoSegmentStream = function(track) {
     boxes.set(moof);
     boxes.set(mdat, moof.byteLength);
 
-    this.trigger('data', boxes);
+    clearDTSInfo(track);
+    this.trigger('data', {track: track, boxes: boxes});
   };
 };
 VideoSegmentStream.prototype = new muxjs.Stream();
 
+collectDTSInfo = function (track, data) {
+  if (track.minDTS === undefined) {
+    track.minDTS = data.dts;
+  } else {
+    track.minDTS = Math.min(track.minDTS, data.dts);
+  }
+
+  if (track.maxDTS === undefined) {
+    track.maxDTS = data.dts;
+  } else {
+    if (data.dts - track.maxDTS > 100) {
+      track.deltaDTS = data.dts - track.maxDTS;
+    }
+    track.maxDTS = Math.max(track.maxDTS, data.dts);
+  }
+};
+
+clearDTSInfo = function (track) {
+  delete track.minDTS;
+  delete track.maxDTS;
+  delete track.deltaDTS;
+};
+
+calculateTrackBMDT = function (track) {
+  var scale;
+  var nextExpectedBMDT = track.nextExpectedBMDT;
+
+  track.baseMediaDecodeTime = track.minDTS - track.earliestDTS;
+
+  if (nextExpectedBMDT) {
+    // We arbitrarily decide that if the difference is greater than one
+    // second we should not attempt to "pull in" the segment in time
```

Comment:
I know this fixes some pretty big issues but I don't think we can use it in general. Unless I'm missing something, this will cause gradual errors in the relationship between the media timeline and PTS values. We need that relationship to stay stable so we can insert timed-metadata at the appropriate positions and seek accurately, for instance.